libkernelflinger/vars.c

```diff
@@ -254,32 +254,72 @@ static void set_provisioning_mode(BOOLEAN provisioning)
 	current_state = provisioning ? UNLOCKED : LOCKED;
 }
 
-enum device_state get_current_state()
+static EFI_STATUS read_efi_device_state(UINT8 *state)
 {
-	UINT8 *stored_state;
-	UINTN dsize;
 	EFI_STATUS ret;
+	UINT8 *stored_state = NULL;
+	UINTN dsize;
 	UINT32 flags;
+
+	ret = get_efi_variable((EFI_GUID *)&fastboot_guid, OEM_LOCK,
+				       &dsize, (void **)&stored_state, &flags);
+	if (EFI_ERROR(ret)) {
+		if (ret == EFI_NOT_FOUND)
+			debug(L"Can not find device state EFI variable");
+		else
+			efi_perror(ret, L"Failed to read device state from EFI variable");
+		return ret;
+	}
+	if (!dsize || !stored_state) {
```

Comment:
`stored_state` cannot be NULL if you arrive here.

Reply:
Just make the code more stable, to avoid NULL point issue without the dependence of the implementation of get_efi_variable().

Reply:
Well `get_efi_variable()` is a Kernelflinger function.  The code you change is also Kernelflinger code => you should use the API as define.  This extra check is unecessary and can confuse the reader.